src/components/Charts/ChartPage.tsx

```diff
@@ -0,0 +1,265 @@
+
+            let temp: { name: string; value: number }[] = monthsArray;
+            for (const meetingFromRes of meetingsFromRes) {
+                if (
+                    temp.some(
+                        (meeting) =>
+                            meeting.name ===
+                            dayjs(meetingFromRes.start_time).format("MMMM")
+                    )
+                ) {
+                    for (const tempVar of temp) {
+                        if (
+                            tempVar.name ===
+                                dayjs(meetingFromRes.start_time).format(
+                                    "MMMM"
+                                ) &&
+                            selectedYear ===
+                                dayjs(meetingFromRes.start_time).year()
+                        )
+                            tempVar.value += 1;
+                    }
+                } else {
+                    temp.push({
+                        name: dayjs(meetingFromRes.start_time).format("MMMM"),
+                        value: 1,
+                    });
+                }
+            }
+            setMeetingsMonth(temp);
+        });
+    };
+
+    const getDaysOfTheWeekFromJson = async () => {
+        await getMeetingsData().then((res) => {
+            const meetingsFromRes = res.data.meetings;
+            let temp: { name: string; value: number }[] = daysArray;
+            for (const meetingFromRes of meetingsFromRes) {
+                if (
+                    temp.some(
+                        (meeting) =>
+                            meeting.name ===
+                            dayjs(meetingFromRes.start_time).format("dddd")
+                    )
+                ) {
+                    for (const tempVar of temp) {
+                        if (
+                            tempVar.name ===
+                                dayjs(meetingFromRes.start_time).format(
+                                    "dddd"
+                                ) &&
+                            selectedWeek ===
+                                Math.ceil(
+                                    dayjs(meetingFromRes.start_time).date() / 7
+                                ) &&
+                            selectedMonth ===
+                                dayjs(meetingFromRes.start_time).format(
+                                    "MMMM"
+                                ) &&
+                            selectedYear ===
+                                dayjs(meetingFromRes.start_time).year()
+                        )
+                            tempVar.value += 1;
+                    }
+                } else {
+                    temp.push({
+                        name: dayjs(meetingFromRes.start_time).format("dddd"),
+                        value: 1,
+                    });
+                }
+            }
+            setMeetingsDay(temp);
+        });
+    };
+
+    useEffect(() => {
```

Comment:
Inside this useEffect in all those function you call getMeetingsData() which is making the same request 3 times, this in not good, try maybe to get the response data from getMeetingsData() and pass the response to the 3 functions